com.ibm.wala.cast.python.ml.test/source/com/ibm/wala/cast/python/ml/test/TestTensorflowModel.java

```diff
@@ -61,44 +63,45 @@ public void testTf1()
 
   @Test
   public void testTf2()
-      throws ClassHierarchyException, IllegalArgumentException, CancelException, IOException {
-    testTf2("tf2.py", "add", 2, 2, 3);
-    testTf2("tf2b.py", "add", 2, 2, 3);
-    testTf2("tf2c.py", "add", 2, 2, 3);
-    testTf2("tf2d.py", "add", 2, 2, 3);
-    testTf2("tf2e.py", "add", 2, 2, 3);
-    testTf2("tf2f.py", "add", 2, 2, 3);
-    testTf2("tf2g.py", "add", 2, 2, 3);
-    testTf2("tf2h.py", "add", 2, 2, 3);
-    testTf2("tf2i.py", "add", 2, 2, 3);
-    testTf2("tf2j.py", "add", 2, 2, 3);
-    testTf2("tf2k.py", "add", 2, 2, 3);
-    testTf2("tf2l.py", "add", 2, 2, 3);
-    testTf2("tf2m.py", "add", 2, 2, 3);
-    // TODO: Uncomment below test when https://github.com/wala/ML/issues/49 is fixed.
-    // testTf2("tf2n.py", "func2", 1, 2);
-    testTf2("tf2o.py", "add", 2, 2, 3);
-    testTf2("tf2p.py", "value_index", 2, 2, 3);
-    testTf2("tf2q.py", "add", 2, 2, 3);
+      throws IllegalArgumentException, CancelException, IOException, WalaException {
+    testTf2("tf2.py", "add", 2, 3, 2, 3);
+    testTf2("tf2b.py", "add", 2, 3, 2, 3);
+    testTf2("tf2c.py", "add", 2, 4, 2, 3);
+    //    testTf2("tf2d.py", "add", 2, 2, 3);
+    //    testTf2("tf2e.py", "add", 2, 2, 3);
+    //    testTf2("tf2f.py", "add", 2, 2, 3);
+    //    testTf2("tf2g.py", "add", 2, 2, 3);
+    //    testTf2("tf2h.py", "add", 2, 2, 3);
+    //    testTf2("tf2i.py", "add", 2, 2, 3);
+    //    testTf2("tf2j.py", "add", 2, 2, 3);
+    //    testTf2("tf2k.py", "add", 2, 2, 3);
+    //    testTf2("tf2l.py", "add", 2, 2, 3);
+    //    testTf2("tf2m.py", "add", 2, 2, 3);
+    //    // TODO: Uncomment below test when https://github.com/wala/ML/issues/49 is fixed.
+    //    // testTf2("tf2n.py", "func2", 1, 2);
+    //    testTf2("tf2o.py", "add", 2, 2, 3);
+    //    testTf2("tf2p.py", "value_index", 2, 2, 3);
+    //    testTf2("tf2q.py", "add", 2, 2, 3);
+    testTf2("tf2r.py", "add", 2, 3, 2, 3);
   }
 
   private void testTf2(
       String filename,
       String functionName,
       int expectedNumberOfTensorParameters,
+      int expectedNumberOfTensorVariables,
       int... expectedValueNumbers)
-      throws ClassHierarchyException, CancelException, IOException {
+      throws CancelException, IOException, WalaException {
```

Comment:
We're probably not going to leave the PDF code here, so this change can ben reverted.

Reply:
Addressed